src/internal/modules/charge-information/lib/charge-information-validator.js

```diff
@@ -37,8 +37,18 @@ const validate = chargeElement =>
 
 const addValidation = chargeInformation => {
   const chargeElementsWithValidationWarnings = chargeInformation.chargeElements.map(element => ({
-    ...element,
-    validationWarnings: validate(element)
+    ...(element.chargePurposes
+      ? {
+        ...element,
+        chargePurposes: element.chargePurposes.map(purpose => ({
+          ...purpose,
+          validationWarnings: validate(purpose)
+        }))
+      }
+      : element),
+    validationWarnings: chargeInformation.scheme === 'alcs'
+      ? validate(element)
+      : [] // a new validator for @SROC will have to be added here
```

Comment:
Please add `todo` in the comment so that SonarCloud will flag it up in the future pull request

Reply:
Info here https://rules.sonarsource.com/go/RSPEC-1135

Reply:
ah nice one i basically removed todo to remove the sonarcloud code smell added it back in.